portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,17 +15,101 @@
 import javax.servlet.http.HttpServletResponse;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comments")
 public class DataServlet extends HttpServlet {
+
+  private Comment comment;
+
+  private final static String ENTITY_NAME = "Comments";
+
+  private final static String NAME_PROPERTY = "name";
+  private final static String COMMENT_PROPERTY = "comment";
+  private final static String TIMESTAMP_PROPERTY = "timestamp";
+  private final static String UPVOTES_PROPERTY = "upvotes";
+  private final static String DOWNVOTES_PROPERTY = "downvotes";
+
+  private final static String ID_PROPERTY = "id";
+
+
+  @Override
+  protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    response.setContentType("application/json");
+    
+    
+    Entity commentEntity = new Entity(ENTITY_NAME);
+
+    String name = request.getParameter("name");
+    String commentText = request.getParameter("comment");
+    long timestamp = System.currentTimeMillis();
+    commentEntity.setProperty(NAME_PROPERTY, name);
+    commentEntity.setProperty(COMMENT_PROPERTY, commentText);
+    commentEntity.setProperty(TIMESTAMP_PROPERTY, timestamp);
+    commentEntity.setProperty(UPVOTES_PROPERTY, 0);
+    commentEntity.setProperty(DOWNVOTES_PROPERTY, 0);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    System.out.println("Added comment to datastore with " + name + " " + commentText + " on " + timestamp);
+
+    response.sendRedirect("/index.html");
+  }
+
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Hisao!");
+    response.setContentType("application/json;");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    ArrayList<Comment> receivedComments = new ArrayList<>();
+    //TODO: Change to sort by up/down vote when that is implemented.
+    Query query = new Query(ENTITY_NAME).addSort(TIMESTAMP_PROPERTY, SortDirection.DESCENDING); 
+    PreparedQuery results = datastore.prepare(query);
+
+    for(Entity entity : results.asIterable()){
+      System.out.println("Got comment from datastore with properties: " + entity.getProperties()); //Debug output
```

Comment:
There is actually a standard way of logging in Java, see [details here](https://cloud.google.com/appengine/docs/standard/java/logs)

Reply:
done!